File: 1290 Convert Binary Number in a Linked List to Integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        if (head == null) return 0;
        int res = head.val;
        while (head.next != null) {
            head = head.next;
            res = 2 * res + head.val;
        }
        return res;
    }
}

File: 1290 Convert Binary Number in a Linked List to Integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        if not head: return 0
        res = head.val
        while head.next:
            head = head.next
            res = 2 * res + head.val
        return res